source/hands_on/ICM.rst

```diff
@@ -0,0 +1,265 @@
+
+Quick Facts
+-------------
+
+1. The baseline reinforcement learning algorithm of ICM is `A3C <http://proceedings.mlr.press/v48/mniha16.pdf>`__ , you can refer to our implementation `A2C <https://github.com/opendilab/DI -engine/blob/main/ding/policy/a2c.py>`__ , if you want to implement A3C, you can use multiple environments to train at the same time.
+
+2. In the follow-up work `Large-Scale Study of Curiosity-Driven Learning <https://arxiv.org/pdf/1808.04355v1.pdf>`__, the baseline algorithm used is PPO, you can refer to our implementation `PPO <https://github.com/opendilab/DI-engine/blob/main/ding/policy/ppo.py>`__, through the PPO algorithm, only a small amount of hyperparameter fine-tuning is required to obtain a robust learning effect .
+
+3. Although both the reward model and the forward model will participate in the calculation of loss, only the forward model will be used as an intrinsic reward. The larger the loss of the forward model, the more inaccurate the estimation of the state characteristics at the next moment based on the current state characteristics and the current action, that is, this state has not been encountered before, and it is worth exploring; the reward model is not an intrinsic reward, its role is mainly to better help characterize the environmental features related to agent actions in the process of feature space extraction.
+
+4. Reward normalization. Since the reward is unstable, it is necessary to normalize the reward to [0, 1] to make the learning more stable. Here we use the maximum and minimum normalization method.
+
+5. Feature normalization. By integrating intrinsic and extrinsic rewards, it is important to ensure that intrinsic rewards scale across different feature representations, which can be achieved through batch normalization.
+
+6. More actors (more collectors in DI-engine): Adding more parallel actors can make training more stable.
+
+Key Equations or Key Graphs
+---------------------------------------
+The overall training and calculation process of the ICM algorithm is as follows:
+
+.. image:: images/ICM_illustraion.png
+   :align: center
+   :scale: 70%
+
+1. As shown in the figure on the left, the agent samples the action a in the state :math:`s_t` through the current policy :math:`\pi` and executes it, and finally obtains the state :math:`s_{t+1}`. The total reward is the sum of two partial rewards, one part is the external reward: math:`r_t^e`, that is, the sparse reward obtained in the environment; the other part is the intrinsic reward obtained by the ICM :math:`r_t^i` (The specific calculation process is given in step 4), the final strategy needs to achieve the purpose of training by optimizing the total reward.

```

Comment:
1. As shown in the figure on the left, the agent samples the action a in the state :math:`s_t` through the current policy :math:`\pi` and executes it, and finally obtains the state :math:`s_{t+1}`. The total reward is the sum of two partial rewards, one part is the external reward :math:`r_t^e`, that is, the sparse reward obtained in the environment; the other part is the intrinsic reward obtained by the ICM :math:`r_t^i` (The specific calculation process is given in step 4), the final strategy needs to achieve the purpose of training by optimizing the total reward.